lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -101,6 +101,15 @@ SmallVector<unsigned> getThreadsPerWarp(Attribute layout) {
   return {};
 }
 
+unsigned getWarpSize(Attribute layout) {
```

Comment:
I think it's better to get the warpSize from the mod, since the layout is usually figured from num_warps and warpSize.

Reply:
True, but you don't always have mod available.
The ModuleOp is only visible in a few files and the majority part of Triton backend need other sources to figure out the current warp size. `Attribute` is the existing way to propagate the property.

Reply:
Yeah. But you can always obtain the mod easily, such as getOperation() and getParentOfType<ModuleOp>().
Why they need other sources to figure out the warpSize? I think it should be determined as early as possible since it is basically arch-specific.

Reply:
@xinyazhang The upstream does not obtain warpSize from layouts. And we should be able to copy what upstream does. Are there any scenarios that we cannot obtain the mod and then the warpSize from the mod?

Reply:
The other method used by upstream is to propagate the warp size through `TritonGPUTypeConverter`, but it's incomplete. Upstream PR https://github.com/openai/triton/pull/1753 is needed to make the remaining warpSize dynamic, but that PR is unmerged yet.

Additionally layout is generally available and the unmerged upstream PR requires more changes on function signatures.

Reply:
I'm fine to merge this PR for now with an agreement to get rid of this getWarpSize-from-layout scheme when upstream's work about dynamic warpSize, including renaming and better propagation, is done.